jane/doc/extensions/modes/reference.md

```diff
@@ -0,0 +1,12 @@
+# Lazy
+`lazy e` contains a thunk that evaluates `e`, as well as a mutable cell to store the
+result of `e`. Upon construction, `lazy e` follows the mode of `e` and the comonadic mode
+of the thunk. Upon destruction (forcing a lazy value), the result follows the mode of lazy
+value. Currently, the thunk, the result and the `lazy` are all heap-allocated.
+
+Additionally, forcing a lazy value involves accessing the mutable cell and thus requires
```

Comment:
This paragraph I understand. But it would probably also be helped by an example.